### PR TITLE
Make TSVConnSslConnectionGet available on QUIC connections

### PR DESCRIPTION
**Pull request by maskit**

### CODE REVIEW

File: src/api/InkAPI.cc [hunk 1]
```diff
@@ -7921,11 +7921,10 @@ TSVConnTunnel(TSVConn sslp)
...
 {
-  TSSslConnection    ssl    = nullptr;
-  NetVConnection    *vc     = reinterpret_cast<NetVConnection *>(sslp);
-  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(vc);
-  if (ssl_vc != nullptr) {
-    ssl = reinterpret_cast<TSSslConnection>(ssl_vc->ssl);
+  TSSslConnection ssl   = nullptr;
+  NetVConnection *netvc = reinterpret_cast<NetVConnection *>(sslp);
+  if (auto tbs = netvc->get_service<TLSBasicSupport>(); tbs) {
+    ssl = reinterpret_cast<TSSslConnection>(tbs->get_tls_handle());
```
Comment (JosiahWI) [inline src/api/InkAPI.cc:7927]: Does this have to be a `reinterpret_cast`?

Comment (maskit) [reply src/api/InkAPI.cc:7927]: Yes.
```
/Users/mkitajo/src/github.com/trafficserver/src/api/InkAPI.cc:7927:11: error: static_cast from 'TLSHandle' (aka 'ssl_st *') to 'TSSslConnection' (aka 'tsapi_ssl_obj *'), which are not related by inheritance, is not allowed
    ssl = static_cast<TSSslConnection>(tbs->get_tls_handle());
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/mkitajo/src/github.com/trafficserver/my-build-quiche/include/ts/apidefs.h:1039:32: note: 'tsapi_ssl_obj' is incomplete
using TSSslConnection = struct tsapi_ssl_obj *;
                               ^
/Users/mkitajo/opt/boringssl/include/openssl/base.h:376:16: note: 'ssl_st' is incomplete
typedef struct ssl_st SSL;
               ^
1 error generated.
```

**Comment by maskit** [comment]
I don't think we state which ones are supported for QUIC connections. Basically all should work once we remove all the dynamic_casts to SSLNetVC.

**Review by JosiahWI — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**